hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongContainer.java

```diff
@@ -16,10 +16,22 @@
 
 package com.hazelcast.concurrent.atomiclong;
 
+import com.hazelcast.config.AtomicLongConfig;
+
 public class AtomicLongContainer {
```

Comment:
Could it be that atomiclongs now consume more memory? So imagine you create 1M atomic longs, then I guess there will be 1M extra AtomicLongConfigs? Afaik new config instance per container.

Reply:
Yes, but we need the config for the merge policies there, like in all other data structure containers.

Reply:
You won't like the split-brain healing implementation for `IAtomicLong`, it'll add more fields to this container ;)

Reply:
It isn't only the field, it will also gets an additional object: the AtomicLongConfig.

Reply:
Yes, but as I said, we need a configuration for the split-brain protection (aka quorum) and split-brain healing (aka merge policies). These are both 3.10 PRDs.

Reply:
We can make a note in the release notes about this. My idea is that this is an issue that needs to be addressed on the level of all config classes - since practically none of them mutate the config, they might as well share the config. I would then add this as a PRD. 
If we introduce immutable configs in 4.0, sharing configuration will be even more safe.

Reply:
If we look at the `Config#findAtomicLongConfig`, which is what the container uses, it will not create new instances of the `AtomicLongConfig` for each container. It should construct one read-only instance which all of the containers will use. 
As Peter has noticed, there seems to be a lack of happens before relationship when publishing the read-only config. The config may be instantiated by different partition threads, depending on the atomic long container name. This is an issue that should be addressed for all structures.

Reply:
This should be not a new issue, since I copied the logic 1:1 from all the other config handling methods. So hopefully it's not blocker for this PR, because we need this to be merged.

Reply:
I agree. I will open an github issue as it is an issue that affects all existing uses of `Config.find***`